test/extended/controller_manager/pull_secret.go

```diff
@@ -175,26 +168,23 @@ var _ = g.Describe("[Feature:OpenShiftControllerManager]", func() {
 		}
 
 		// Expect the matching dockercfg secret to also be deleted
-		waitForSecretDelete(dockercfgSecretName, secretsWatch, t)
-	})
-})
-
-func waitForSecretDelete(secretName string, w watch.Interface, t g.GinkgoTInterface) {
-	for {
-		select {
-		case event := <-w.ResultChan():
-			secret, ok := event.Object.(*corev1.Secret)
-			if !ok {
-				t.Fatalf("Got an event that was not a secret: %v %#v", event.Type, event.Object)
-			}
-			secret.Data = nil // reduce noise in log
-			t.Logf("%s got %s %s %#v", time.Now().Format(time.RFC3339Nano), event.Type, secret.ResourceVersion, secret)
-			if event.Type == watch.Deleted && secret.Name == secretName {
-				return
+		timeout := time.NewTimer(5 * time.Minute)
+		ticker := time.NewTicker(5 * time.Second)
+		defer timeout.Stop()
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-timeout.C:
+				t.Fatalf("timeout waiting for secret deletion")
+			case <-ticker.C:
+				if _, err := clusterAdminKubeClient.CoreV1().Secrets(sa.Namespace).Get(
+					dockercfgSecretName,
+					metav1.GetOptions{},
+				); errors.IsNotFound(err) {
+					return
+				}
```

Comment:
nit: use `wait.Poll` from `k8s.io/apimachinery/pkg/util/wait`. It effectively does the same thing, returning a specific timeout error if it is exceeded.

See https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L65-L189 as an example.

Reply:
Done, PTAL.